subprojects/GOPATH/src/github.com/microsoft/hivedscheduler/pkg/algorithm/types.go

```diff
@@ -153,3 +154,41 @@ func (aag *AlgoAffinityGroup) ToAffinityGroup() api.AffinityGroup {
 	ag.Status.LazyPreemptionStatus = aag.lazyPreemptionStatus
 	return ag
 }
+
+type affinityGroupPlacement map[int32][]CellList
+
+func physicalPlacementToString(placement affinityGroupPlacement) string {
+	nodeToGpuIndices := map[string][]int32{}
```

Comment:
Seems you can use 
func (s affinityGroupPlacement) ToString() string ?

Reply:
Use different type to differentiate them?
type physicalPlacement map[int32][]CellList
type virtualPlacement map[int32][]CellList


---
In reply to: [393421975](https://github.com/microsoft/pai/pull/4262#discussion_r393421975) [](ancestors = 393421975)